File: app/src/main/java/com/byd/dashcast/proxy/MirrorResourceOwner.java
package com.byd.dashcast.proxy;

import android.os.Binder;
import android.os.IBinder;

/** Process-lifetime token whose Binder death lets MirrorDaemon release transient mirror state. */
public final class MirrorResourceOwner {

    private static final IBinder TOKEN = new Binder();

    private MirrorResourceOwner() {}

    public static IBinder token() {
        return TOKEN;
    }
}

File: app/src/main/java/com/byd/dashcast/proxy/MirrorResourceOwner.kt
package com.byd.dashcast.proxy

import android.os.Binder
import android.os.IBinder

/** Process-lifetime token whose Binder death lets the SurfaceDaemon release transient mirror state. */
object MirrorResourceOwner {

    private val TOKEN: IBinder = Binder()

    @JvmStatic
    fun token(): IBinder = TOKEN
}
